lib/makara/cookie.rb

```diff
@@ -0,0 +1,50 @@
+module Makara
+  module Cookie
+    extend self
+
+    IDENTIFIER = '_mkra_stck'.freeze
+
+    DEFAULT_OPTIONS = {
+      path: "/",
+      http_only: true
+    }
+
+    def fetch(request)
+      parse(request.cookies[IDENTIFIER].to_s)
+    end
+
+    def store(context_data, headers, options = {})
+      unless context_data.nil?
```

Comment:
> this could be nil if there were no changes to the cookie 
Correct!

> Do you have to set the header here to keep the cookie around?

No! If you don't send anything the cookie will be kept as it was. That was my idea of returning `nil` here: there is no _next_ context then there is no cookie sent.

> In which case, we do we delete the cookie in some case?
We do! When the _next_ context exists and it's empty, we send a cookie with `max-age=0` and `expires=<current time>`. That's a way of _deleting_ the cookie.

Reply:
I see. thanks.